components/slideshow.tsx

```diff
@@ -0,0 +1,97 @@
+'use client';
+import Image from 'next/image';
+import { useCallback, useEffect, useState } from 'react';
+import { FaChevronLeft, FaChevronRight } from 'react-icons/fa';
+
+export default function Carousel({ images }: { images: string[] }) {
+  const [currentSlide, setCurrentSlide] = useState<number>(0);
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  const prevSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === 0 ? images.length - 1 : prev - 1
+    );
+  }, [images.length]);
+
+  const nextSlide = useCallback(() => {
+    setCurrentSlide((prev: number) =>
+      prev === images.length - 1 ? 0 : prev + 1
+    );
+  }, [images.length]);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      nextSlide();
+    }, 3000);
+
+    return () => clearInterval(intervalId);
+  }, [nextSlide]);
+
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === 'ArrowLeft') {
+        prevSlide();
+      } else if (event.key === 'ArrowRight') {
+        nextSlide();
+      }
+    },
```

Comment:
This should only execute if the slideshow is hovered on.
```suggestion
    (event: KeyboardEvent) => {
      if (!hovered) return;

      if (event.key === 'ArrowLeft') {
        prevSlide();
      } else if (event.key === 'ArrowRight') {
        nextSlide();
      }
    },
```